pydropsonde/helper/rawreader.py

```diff
@@ -40,69 +40,61 @@ def check_launch_detect_in_afile(a_file: "str") -> bool:
     bool
         True if launch is detected (1), else False (0)
     """
+    if a_file is not None and os.path.getsize(a_file) > 0:
+        with open(a_file, "r") as f:
+            module_logger.debug(f"Opened File: {a_file=}")
+            lines = f.readlines()
 
-    with open(a_file, "r") as f:
-        module_logger.debug(f"Opened File: {a_file=}")
-        lines = f.readlines()
-
-        for i, line in enumerate(lines):
-            if "Launch Obs Done?" in line:
-                line_id = i
-                module_logger.debug(f'"Launch Obs Done?" found on line {line_id=}')
-                break
-
-        return bool(int(lines[line_id].split("=")[1]))
+            for i, line in enumerate(lines):
+                if "Launch Obs Done?" in line:
+                    line_id = i
+                    module_logger.debug(f'"Launch Obs Done?" found on line {line_id=}')
+                    break
 
+            return bool(int(lines[line_id].split("=")[1]))
```

Comment:
This still uses `line_id` from inside the for loop, which might lead to an unbound local error. Instead, it probably would be more elegant to directly return from within the loop an use `line` instead of `lines[line_id]`.